tests/unit/bokeh/core/test_core_serialization.py

```diff
@@ -0,0 +1,907 @@
+        v0 = SomeModel(p0=3, p1="b", p2=[4, 5, 6])
+        val = SomeDataClass(f0=2, f1=[1, 2, 3], f5=v0)
+        encoder = Serializer()
+        rep = encoder.encode(val)
+        assert rep == dict(
+            type="test_core_serialization.SomeDataClass",
+            attributes=dict(
+                f0=2,
+                f1=[1, 2, 3],
+                f2=None,
+                f5=dict(
+                    type="test_core_serialization.SomeModel",
+                    id=v0.id,
+                    attributes=dict(
+                        p0=3,
+                        p1="b",
+                        p2=[4, 5, 6],
+                    ),
+                ),
+            ),
+        )
+        assert encoder.buffers == []
+
+    def test_color_rgb(self) -> None:
+        val = RGB(16, 32, 64)
+        encoder = Serializer()
+        rep = encoder.encode(val)
+        assert rep == "rgb(16, 32, 64)"
+        assert encoder.buffers == []
+
+    def test_color_rgba(self) -> None:
+        val = RGB(16, 32, 64, 0.1)
+        encoder = Serializer()
+        rep = encoder.encode(val)
+        assert rep == "rgba(16, 32, 64, 0.1)"
+        assert encoder.buffers == []
+
+    def test_pd_series_base64(self, pd) -> None:
+        encoder = Serializer(binary=False)
+        val = pd.Series([0, 1, 2, 3, 4, 5], dtype="int32")
+        rep = encoder.encode(val)
+        assert rep == dict(
+            type="ndarray",
+            array=dict(
+                type="bytes",
+                data="AAAAAAEAAAACAAAAAwAAAAQAAAAFAAAA",
+            ),
+            order=sys.byteorder,
+            shape=[6],
+            dtype="int32",
+        )
+        assert encoder.buffers == []
+
+    def test_np_int64(self) -> None:
+        encoder = Serializer()
+        val = np.int64(1).item()
+        rep = encoder.encode(val)
+        assert rep == 1
+        assert isinstance(rep, int)
+
+    def test_np_float64(self) -> None:
+        encoder = Serializer()
+        val = np.float64(1.33)
+        rep = encoder.encode(val)
+        assert rep == 1.33
+        assert isinstance(rep, float)
+
+    def test_np_bool(self) -> None:
+        encoder = Serializer()
+        val = np.bool_(True)
+        rep = encoder.encode(val)
+        assert rep == True
+        assert isinstance(rep, bool)
+
+    def test_np_datetime64(self) -> None:
+        encoder = Serializer()
+        val = np.datetime64('2017-01-01')
+        rep = encoder.encode(val)
+        assert rep == 1483228800000.0
+        assert isinstance(rep, float)
+
+    def test_dt_time(self) -> None:
+        encoder = Serializer()
+        val = dt.time(12, 32, 15)
+        rep = encoder.encode(val)
+        assert rep == 45135000.0
+        assert isinstance(rep, float)
+
+    def test_rd_relativedelta(self) -> None:
+        encoder = Serializer()
+        val = rd.relativedelta()
+        rep = encoder.encode(val)
+        assert isinstance(rep, dict)
```

Comment:
Sure, just note that tests are far from being ready and I'm not sure about serialization of `relativedelta` at all, because currently it doesn't preserve type information.

Reply:
I'm not sure where `relativedelta` would be used with the current serialized representation. We don't have a single test or example for this.

Reply:
AFAIK it should behave the same as a stdlib `datetime.timedelta` But if it doesn't work (and has never worked) let's just remove it.